p2p/chain.py

```diff
@@ -713,19 +714,10 @@ def __init__(self, db: 'AsyncHeaderDB', logger: TraceLogger, token: CancelToken)
             )
 
         limit = max(max_headers, eth.MAX_HEADERS_FETCH)
```

Comment:
This should actually be `min(max_headers, eth.MAX_HEADERS_FETCH)`

Maybe add a test to ensure we return the requested `max_headers`. That is important because currently when geth asks us for the DAO fork block header we respond with 192 headers and it doesn't like that

Reply:
> This should actually be min(max_headers, eth.MAX_HEADERS_FETCH

fix inbound...

> Maybe add a test to ensure we return the requested max_headers.

I'm not really sure where to start on writing that test. It seems like it belongs in `tests/p2p/test_sync.py` somewhere, but that reads a full sync integration test. I didn't see anything low-level like:
- trigger some request in the client
- assert that request received in the server

Any suggestions?

Reply:
Needed some refactoring, but pretty straightforward: #1051